# --- T2-COPYRIGHT-NOTE-BEGIN ---
# This copyright note is auto-generated by ./scripts/Create-CopyPatch.
# 
# T2 SDE: package/.../chrony/gcc4.patch
# Copyright (C) 2004 - 2005 The T2 SDE Project
# 
# More information can be found in the files COPYING and README.
# 
# This patch file is dual-licensed. It is available under the license the
# patched project is licensed under, as long as it is an OpenSource license
# as defined at http://www.opensource.org/ (e.g. BSD, X11) or under the terms
# of the GNU General Public License as published by the Free Software
# Foundation; either version 2 of the License, or (at your option) any later
# version.
# --- T2-COPYRIGHT-NOTE-END ---
--- chrony-1.21/addrfilt.c.orig	2005-08-11 22:32:54.000000000 +0200
+++ chrony-1.21/addrfilt.c	2005-09-04 11:05:54.000000000 +0200
@@ -45,7 +45,7 @@
 
 struct _TableNode;
 
-typedef struct _TableNode ExtendedTable[TABLE_SIZE];
+typedef struct _TableNode *ExtendedTable;
 
 typedef enum {DENY, ALLOW, AS_PARENT} State;
 
@@ -124,7 +124,7 @@
 
   if (node->extended == NULL) {
 
-    node->extended = MallocNew(ExtendedTable);
+    node->extended = MallocArray(ExtendedTable, TABLE_SIZE);
 
     for (i=0; i<TABLE_SIZE; i++) {
       child_node = &((*(node->extended))[i]);
